### ISSUE TITLE
[BUG] MariaDB ORDER BY alias always returns 100 (or fewer) results

### ISSUE THREAD

**Issue by ptyork** [cweijan/vscode-mysql#1046]
### Database type and version:

replicated on:

   * MariaDB 10.6.4
   * MariaDB 11.2.3

unsure of others, or if it also impacts MySQL.

### Description

For some reason, the client is reporting a fixed maximum of 100 results (with no pagination) under some pretty specific circumstances.

These are using the MySQLTutorial "ClassicModels" sample database.

Correctly Shows 2996 Results:
```
SELECT productName, (priceEach - buyPrice) as Profit
FROM products 
NATURAL JOIN orderdetails 
ORDER BY (priceEach - buyPrice);
```
Incorrectly Shows 100 Results with Pagination Disabled:
```
SELECT productName, (priceEach - buyPrice) as Profit
FROM products 
NATURAL JOIN orderdetails 
ORDER BY Profit;
```
These queries are identical and should behave identically. They return the same results in the terminal-based client, but are not behaving as expected in the extension.

I tried a lot of permutations. It seems to require a) at least one join of some sort and b) ordering by the alias. Even the following simple example gets only 100 results.
```
SELECT productName as pname
FROM products 
NATURAL JOIN orderdetails 
ORDER BY pname;
```
Tested on Windows in VSCode all running locally AND in-browser using a Linux Codespace/DevContainer hosted by GitHub. Tested using both MySQL and MariaDB connections, though the ability to install and use the official MariaDB lib appears not to function in either environment.

**Comment by cweijan** [comment]
Thanks feedback. The currently generated count SQL cannot cover all SQL. I will support it in the next version.

**Comment by cweijan** [comment]
Fixed in version 7.0.5.

**Comment by ptyork** [comment]
Confirmed. And thanks for the quick turnaround!! I use this for a college course and it was confusing my students, so the fast fix really helps!
